libraries/rush-lib/src/logic/operations/test/BuildPlanPlugin.test.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+jest.mock('@rushstack/terminal', () => {
+  const originalModule = jest.requireActual('@rushstack/terminal');
+  return {
+    ...originalModule,
+    ConsoleTerminalProvider: {
+      ...originalModule.ConsoleTerminalProvider,
+      supportsColor: true
+    }
+  };
+});
+import { MockWritable, Terminal } from '@rushstack/terminal';
+import { BuildPlanPlugin } from '../BuildPlanPlugin';
+import {
+  type ICreateOperationsContext,
+  type IExecuteOperationsContext,
+  PhasedCommandHooks
+} from '../../../pluginFramework/PhasedCommandHooks';
+import { CollatedTerminalProvider } from '../../../utilities/CollatedTerminalProvider';
+import { CollatedTerminal, StreamCollator } from '@rushstack/stream-collator';
+import type { Operation } from '../Operation';
+import path from 'path';
+import { RushConfiguration } from '../../../api/RushConfiguration';
+import {
+  CommandLineConfiguration,
+  type IPhase,
+  type IPhasedCommandConfig
+} from '../../../api/CommandLineConfiguration';
+import type { ICommandLineJson } from '../../../api/CommandLineJson';
+import { JsonFile } from '@rushstack/node-core-library';
+import { OperationExecutionRecord } from '../OperationExecutionRecord';
+import { PhasedOperationPlugin } from '../PhasedOperationPlugin';
+import type { RushConfigurationProject } from '../../../api/RushConfigurationProject';
+import { RushConstants } from '../../RushConstants';
+import { MockOperationRunner } from './MockOperationRunner';
+import { ProjectChangeAnalyzer } from '../../ProjectChangeAnalyzer';
+
+const mockWritable: MockWritable = new MockWritable();
+const mockTerminal: Terminal = new Terminal(new CollatedTerminalProvider(new CollatedTerminal(mockWritable)));
+
+const mockStreamWritable: MockWritable = new MockWritable();
+const streamCollator = new StreamCollator({
+  destination: mockStreamWritable
+});
+
+describe('BuildPlanPlugin', () => {
+  const rushJsonFile: string = path.resolve(__dirname, `../../test/workspaceRepo/rush.json`);
+  const commandLineJsonFile: string = path.resolve(
+    __dirname,
+    `../../test/workspaceRepo/common/config/rush/command-line.json`
+  );
+  let rushConfiguration!: RushConfiguration;
+  let commandLineConfiguration!: CommandLineConfiguration;
+
```

Comment:
```suggestion
describe('BuildPlanPlugin', () => {
  const rushJsonFile: string = path.resolve(__dirname, `../../test/workspaceRepo/rush.json`);
  const commandLineJsonFile: string = path.resolve(
    __dirname,
    `../../test/workspaceRepo/common/config/rush/command-line.json`
  );
  let rushConfiguration!: RushConfiguration;
  let commandLineConfiguration!: CommandLineConfiguration;
  let stringBufferTerminalProvider!: StringBufferTerminalProvider;
  let terminal!: Terminal;
  const mockStreamWritable: MockWritable = new MockWritable();
  const streamCollator = new StreamCollator({
    destination: mockStreamWritable
  });

  beforeEach(() => {
    stringBufferTerminalProvider = new StringBufferTerminalProvider();
    terminal = new Terminal(stringBufferTerminalProvider);
    mockStreamWritable.reset();
  })

```

Reply:
Updated! Thoughts on the snapshot? I think it's standard for the `StringBufferTerminalProvider`, but IMO it doesn't look as useful as the previous terminal snapshot (all the `[n]`)

Reply:
https://github.com/microsoft/rushstack/pull/4660#discussion_r1614224802